impeller/compiler/shader_lib/impeller/prefix_sum.glsl

```diff
@@ -0,0 +1,38 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Parallel prefix sum computes the prefix in place in storage.
+// BLOCK_SIZE is the overall storage size while ident must be the global
+// x identifier.
+#define PrefixSum(ident, storage, BLOCK_SIZE)      \
```

Comment:
This is a simple and not particularly fast implementation, but its good enough to start with and could be swapped for a more efficient implementation later.

Reply:
I was thinking we could if/def or function constant our way into more efficient implementations. I have more reading to do though. Where should I look into the subgroupInclusive/ExclusiveAdd?